backend/backend/game.py

```diff
@@ -1,47 +1,173 @@
-"""Module to create game objects"""
+"""This module provides the Game class"""
 
-import uuid
+from operator import itemgetter
+from itertools import groupby
 
+import backend.storage
 
-class Monopoly:  # pylint: disable=too-few-public-methods
-    """A class representing a game of monopoly.
-        TODO: Add game to storage.
-    >>> print(Monopoly(4)) # doctest: +ELLIPSIS
-    Game id:...
-    Lobby State:0/4
-    Game Ready:False
-    Players:[]
-    <BLANKLINE>
+
+class Game(object):
+    """A single game of monopoly. Refer to the Player class for how to
+    access and mutate members."""
+    def __init__(self, uid):
+        self._uid = uid
+        self._in_context = False
+        self._players = None
+        self._current_turn = None
+        self._state = None
+        self._conn = None
+
+    def __enter__(self):
+        self._in_context = True
+        self._conn = backend.storage.make_connection()
+        self._conn.begin()
+        with self._conn.cursor() as cursor:
+            cursor.execute('SELECT * FROM `games` WHERE `id` = %s;',
+                           (self.uid,))
+            result = cursor.fetchone()
+            self._current_turn = result['current_turn']
+            self._state = result['state']
+            del result
+            cursor.execute('SELECT `player_id` FROM `playing_in` '
+                           'WHERE `game_id` = %s;',
+                           (self.uid,))
+            self._players = [result['player_id']
+                             for result in cursor.fetchall()]
+        return self
+
+    def __exit__(self, *exc):
+        try:
+            with self._conn.cursor() as cursor:
+                cursor.execute('UPDATE `games` '
+                               'SET `current_turn` = %s, '
+                               '`state` = %s, '
+                               'WHERE `id` = %s;',
+                               (self.current_turn, self.state,
+                                self.uid))
+                cursor.execute('DELETE FROM `playing_in` WHERE `id = %s;',
```

Comment:
Didn’t notice this till now, but is there a problem here? The backtick before `id = %s;',` doesn’t get closed until the backtick before `playing_in` on line 49, which I don’t think could be right.

(I’m not actually sure what the backtick does here.)